docs/blog/arcade/tilecode.md

```diff
@@ -0,0 +1,59 @@
+
+TileCode enables the process of game creation to take place on gaming handhelds themselves, rather than tablets/laptops/desktops, using just the **four-way direction pad** and the **A and B buttons** common to most handhelds,
+as shown above. With TileCode, we’ve already created a variety of retro video games, including Snake, Bejeweled, Pac-Man, Boulder Dash, and Sokoban (see examples in the banner).
+
+### TileCode: Design and Coding
+
+As shown on the TileCode home screen (below-left), there are eight game slots available to program (on an Arcade device, game assets are stored in the device's non-volatile flash memory; in the browser, game assets are stored in browser local storage). For each game, TileCode allows users to select game characters (sprites) and game background (tiles) from a gallery, modify the sprites and tile backgrounds, and create a game level by editing the game map, as shown in the three other screens:
+
+![TileCode screens](/static/blog/arcade/tilecode/screens.JPG)
+
+A TileCode program is a set of rules, each of which is associated with a sprite. A rule takes the form of a **When-Do** pair, as shown below on the left (screen labeled "code"). The **When** section visually describes a pattern/predicate over the 3x3 local neighborhood around the central sprite (the player sprite, in this case) to be matched against the tile map.  The **Do** section contains commands that are sent to the identified sprite when the pattern matches.
+
+![When-Do Rule](/static/blog/arcade/tilecode/editPlayMap.JPG)
+
+The rule above fires when the user presses the right-dpad button, and there is a player sprite on the tile map, and there is grass on the tile to the right of the player, sending the player sprite a move-right command.
+
+From this rule example we see how TileCode encourages the user to explore the relationship between the data on the map and how the rules fire based on the patterns present in this map. In the rule coding screen (shown above), the user can play the game, return to the coding screen to change the rule and see its effect on game play. She can also visit the tile map editor and make changes to enable or disable the firing of a rule. (Many beginner game creation environments use block-based programming that emphasizes the **structured control-flow** constructs of modern programming languages. In contrast, TileCode uses **cellular automata** as its basic programming model, in which parallel pattern matching against the tile map is the main focus).
```

Comment:
```suggestion
From this rule example we see how TileCode encourages the user to explore the relationship between the data on the map and how the rules fire based on the patterns present in this map. In the rule coding screen (shown above), the user can play the game, return to the coding screen to change the rule, and see its effect on game play. She can also visit the tile map editor and make changes to enable or disable the firing of a rule. (Many beginner game creation environments use block-based programming that emphasizes the **structured control-flow** constructs of modern programming languages. In contrast, TileCode uses **cellular automata** as its basic programming model, in which parallel pattern matching against the tile map is the main focus).
```

Reply:
There's only a comma inserted in `` to change the rule, and`` but the diff isn't showing.